shopfloor_base_default_profile/services/user.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 Brian McMaster <brian@mcmpest.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+from odoo.addons.component.core import Component
+
+class ShopfloorUser(Component):
+    _inherit ="shopfloor.user"
+
+    @property
+    def _user_info_parser(self):
+        res = super()._user_info_parser
+        res.append(
+            ("shopfloor_default_profile_ids:default_profile", self._user_profile_parser)
+        )
+        return res
+    
+    def _user_profile_parser(self, rec, fname):
+        profile = rec[fname].search([("shopfloor_app_id", "=", self.collection.id)], limit=1).profile_id
```

Comment:
There's no need to search IMO.
```suggestion
        profile = rec[fname].filtered(lambda x: x.shopfloor_app_id == self.collection.id).profile_id
```